test/controller.test.ts

```diff
@@ -530,6 +536,645 @@ contract('Controller', ([owner, accountOwner1, accountOperator1, random]) => {
+        collateralToDeposit.toString(),
+        'Collateral asset amount deposited into vault mismatch',
+      )
+    })
+
+    it('should deposit a whitelisted collateral asset from account operator', async () => {
+      const vaultCounter = new BigNumber(await controller.getAccountVaultCounter(accountOwner1))
+      assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
+
+      const collateralToDeposit = new BigNumber('10')
+      const actionArgs = [
+        {
+          actionType: ActionType.DepositCollateral,
+          owner: accountOwner1,
+          sender: accountOperator1,
+          asset: usdc.address,
+          vaultId: vaultCounter.toNumber(),
+          amount: collateralToDeposit.toNumber(),
+          index: '0',
+          data: ZERO_ADDR,
+        },
+      ]
+
+      const marginPoolBalanceBefore = new BigNumber(await usdc.balanceOf(marginPool.address))
+      const senderBalanceBefore = new BigNumber(await usdc.balanceOf(accountOperator1))
+      const vaultBefore = await controller.getVault(accountOwner1, vaultCounter)
+
+      await usdc.approve(marginPool.address, collateralToDeposit, {from: accountOperator1})
+      await controller.operate(actionArgs, {from: accountOperator1})
+
+      const marginPoolBalanceAfter = new BigNumber(await usdc.balanceOf(marginPool.address))
+      const senderBalanceAfter = new BigNumber(await usdc.balanceOf(accountOperator1))
+      const vaultAfter = await controller.getVault(accountOwner1, vaultCounter)
+
+      assert.equal(
+        marginPoolBalanceAfter.minus(marginPoolBalanceBefore).toString(),
+        collateralToDeposit.toString(),
+        'Margin pool balance collateral asset balance mismatch',
+      )
+      assert.equal(
+        senderBalanceBefore.minus(senderBalanceAfter).toString(),
+        collateralToDeposit.toString(),
+        'Sender balance collateral asset balance mismatch',
+      )
+      assert.equal(vaultAfter.collateralAssets.length, 1, 'Vault collateral assets array length mismatch')
+      assert.equal(
+        vaultAfter.collateralAssets[0],
+        usdc.address,
+        'Collateral asset address deposited into vault mismatch',
+      )
+      assert.equal(
+        new BigNumber(vaultAfter.collateralAmounts[0])
+          .minus(new BigNumber(vaultBefore.collateralAmounts[0]))
+          .toString(),
+        collateralToDeposit.toString(),
+        'Long otoken amount deposited into vault mismatch',
+      )
+    })
+
+    it('should revert depositing a collateral asset from a msg.sender different than arg.from', async () => {
+      const vaultCounter = new BigNumber(await controller.getAccountVaultCounter(accountOwner1))
+      assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
+
+      const collateralToDeposit = new BigNumber('10')
+      const actionArgs = [
+        {
+          actionType: ActionType.DepositCollateral,
+          owner: accountOwner1,
+          sender: random,
```

Comment:
we should test this specifically for the vault operator as the sender and the user as the sender as well, depending on what decision we make there.

Reply:
random can be any address, right ?

Reply:
yes, but the reason i say we should test for the vault operator and vault owner is that if we decide that vault operator should be able to take money from vault owner, then that is something we need to ensure we test. If not, we need to test that its not possible.